lib/queuePopulator/IngestionProducer.js

```diff
@@ -381,6 +399,28 @@ class IngestionProducer {
         }
         return done(null, []);
     }
+
+    /**
+     * Get bucket cseq
+     * @param {string} bucket - bucket name
+     * @param {function} done - callback(err, cseq) where `cseq` is an integer
+     * @return {undefined}
+     */
+    _getBucketCseq(bucket, done) {
+        return this._ringReader.getBucketCseq({ Bucket: bucket },
+        (err, data) => {
+            if (err || !data[0] || !data[0].cseq) {
```

Comment:
In what case would an error be returned? Could it ever by worthwhile to retry this request before declaring it an internal error?

Reply:
Definitely, I agree, we need to make the service more durable adding retries where applicable (especially on processor side).

Probably the most common errors here will be the 404/500 errors if the source is down or having problems. Possible that bucket may not exist, but this should be caught prior to this phase